frontend/src/components/uam/components/users-table/users-table.tsx

```diff
@@ -15,11 +16,24 @@ import { uamActions } from 'store/actions';
 import styles from './styles.module.scss';
 
 const UsersTable: FC = () => {
-  const [page, setPage] = useState<number>(PaginationDefaultValue.DEFAULT_PAGE);
+  const [searchParams, setSearchParams] = useSearchParams();
+
+  const getValidPageFromSearchParams = (): number => {
+    const pageFromQuery = Number(searchParams.get('page'));
```

Comment:
create a `usePagination` hook that will handle all pagination

we need to apply it to other tables